nixos/modules/tasks/network-interfaces.nix

```diff
@@ -395,7 +395,16 @@ in
 
     networking.nameservers = mkOption {
       type = types.listOf types.str;
-      default = [];
+      default =
```

Comment:
no, option description shouldn't have complex logic - this is stuff rendered for docs too.

Reply:
Always add "127.0.0.1" without possibility to revoke only because ```service.unbound.enable``` is true  - as the thinks work now - it bad too

Reply:
Or did you just mean to move the logic upwards into let-block?

Reply:
I mean that while logic is OK, docs will show `default = [];`. So this should be moved out to `lib.mkDefault` in implementation section.

Also I think it will break existing deployments, those that enable both local resolver and set `networking.nameservers` - 127.0.0.1 will get lost after this change.

Reply:
> Also I think it will break existing deployments, those that enable both local resolver and set networking.nameservers - 127.0.0.1 will get lost after this change.

127.0.0.1 was lost anyway [in ```network-setup.service```](https://github.com/volth/nixpkgs/compare/55fa156...danbst:patch-13#diff-0346c64733c3dc6b8b853e265bb1e4f1L105)

I was exactly one who has local resolver enabled (although the resolver listened on an external interface only and did not listen on ```127.0.0.1```) and set networking.nameservers to, say, ```["8.8.8.8"]```

So ```/etc/resolv.conf``` had "nameserver 8.8.8.8" on the server boot which turned into "nameserver 127.0.0.1" on ```nixos-rebuild```

Breaking such existing deployments is exactly the goal :)